db/models/articles.models.js

```diff
@@ -14,6 +15,22 @@ function fetchArticleById(article_id){
         }
           return article
         })
-}
 
-module.exports = { fetchArticleById }
\ No newline at end of file
+}
+    exports.fetchAllArticles = ()=>{
+        const queryStatment = `
+        SELECT articles.*, COUNT(comments.article_id) FROM articles
+        LEFT JOIN comments ON articles.article_id = comments.article_id
+        GROUP BY (articles.article_id)
+        ORDER BY article_id DESC;`
+        return db
+        .query(queryStatment)
+        .then(({rows})=> {
+            for(let articleIndex = 0; articleIndex<rows.length; articleIndex++){
+                const currentArticle = rows[articleIndex]
+                currentArticle.comment_count = parseInt(currentArticle.count)
+                delete currentArticle.count
+            }
+              return rows
```

Comment:
I see what you are doing here, but it isn't needed. It is fine for the comment count to be a string. But if you really want it to be a number, you can get SQL to coerce it into an INT for you. You would need to research how to do this, but honestly, not needed 😄